### PR TITLE
sqlcipher: enable JSON1 extension by default

### PR DESCRIPTION
**Pull request by justinclift**
Also added a test - copied & modified from the SQLite formula. :)

### CODE REVIEW

**Comment by justinclift** [comment]
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

File: Formula/sqlcipher.rb [hunk 1]
```diff
@@ -34,13 +34,27 @@ def install
...
     system "make", "install"
   end
+
+  test do
+    path = testpath/"school.sql"
+    path.write <<-EOS.undent
+      create table students (name text, age integer);
+      insert into students (name, age) values ('Bob', 14);
+      insert into students (name, age) values ('Sue', 12);
+      insert into students (name, age) values ('Tim', 13);
```
Comment (wmertens) [inline Formula/sqlcipher.rb]: would be good to add a test for JSON1 extension:
```
insert into students (name, age) values ('Tim', json_extract('{"age": 13}', '$.age'));
```

**Comment by justinclift** [comment]
Oh, good thinking.  I'll add that in a sec.  Thanks @wmertens. :smile:

**Comment by justinclift** [comment]
Just added that to the PR.  Note that I did that blind, as I'm on a Linux desktop atm not my OSX one.  Hopefully got it right. :wink:

**Comment by justinclift** [comment]
@ilovezfs Errr... ping?

File: Formula/sqlcipher.rb [hunk 2]
```diff
@@ -34,13 +34,27 @@ def install
     ]
 
     if build.with?("fts")
-      args << "CFLAGS=-DSQLITE_HAS_CODEC -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5"
+      args << "CFLAGS=-DSQLITE_HAS_CODEC -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5"
```
Comment (ilovezfs) [inline Formula/sqlcipher.rb]: If it's enabled in both the if and the else, then it belongs in the array above :)

**Review by ilovezfs — DISMISSED** [review]

File: Formula/sqlcipher.rb (hunk 2 above)
Comment (justinclift) [reply Formula/sqlcipher.rb]: Oops.  Good point.  Will fix shortly, thanks. :smile:

Comment (justinclift) [inline Formula/sqlcipher.rb]: Hmmm, so is the existing `-DSQLITE_HAS_CODEC`.  Might as well put that in the array above too...

Comment (ilovezfs) [reply Formula/sqlcipher.rb]: Hmm I guess we can't since this is actually a single argument. :see_no_evil:

**Comment by ilovezfs** [comment]
Thanks @justinclift! Shipped! :ship:

As an aside: please avoid using @personsusername in commit messages so they don't get spammed by the bots :)

File: Formula/sqlcipher.rb (hunk 2 above)
Comment (justinclift) [reply Formula/sqlcipher.rb]: Heh Heh Heh, just switched to my OSX desktop to get this done... and yeah.  Oops. :smile:

**Comment by justinclift** [comment]
> As an aside: please avoid using @personsusername in commit messages so they don't get spammed by the bots :)

Ahhh.  Didn't realise that, makes sense.  Thanks @ilovezfs. :smile:

**Pull request closed without merging**